### PR TITLE
SFTP implementation

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**

### CODE REVIEW

File: src/ssh.c [hunk 1]
```diff
@@ -608,6 +620,13 @@ int wolfSSH_TriggerKeyExchange(WOLFSSH* ssh)
     return ret;
 }
 
+#if 0
```
Comment (ejohnstown) [inline src/ssh.c]: Can this be removed? We also have DumpOctetString available in internal.c.

File: examples/echoserver/echoserver.c [hunk 2]
```diff
@@ -95,74 +96,118 @@ static int dump_stats(thread_ctx_t* ctx)
...
+static THREAD_RETURN WOLFSSH_THREAD server_worker(void* vArgs)
+{
+    int ret;
+    thread_ctx_t* threadCtx = (thread_ctx_t*)vArgs;
+
+    if (wolfSSH_accept(threadCtx->ssh) == WS_SUCCESS) {
+        const char* cmd = wolfSSH_GetSessionCommand(threadCtx->ssh);
+
+        /* handle if is SFTP channel */
+        if (WOLFSSH_SESSION_SUBSYSTEM == wolfSSH_GetSessionType(threadCtx->ssh)
```
Comment (ejohnstown) [inline examples/echoserver/echoserver.c]: Could this be a `strcpy()` or `strncpy()`?

File: src/ssh.c [hunk 3]
```diff
@@ -383,6 +383,17 @@ int wolfSSH_accept(WOLFSSH* ssh)
             ssh->acceptState = ACCEPT_CLIENT_SESSION_ESTABLISHED;
             WLOG(WS_LOG_DEBUG, acceptState, "CLIENT_SESSION_ESTABLISHED");
 
+        #ifdef WOLFSSH_SFTP
+        case ACCEPT_CLIENT_SESSION_ESTABLISHED:
+        {
+            const char* cmd = wolfSSH_GetSessionCommand(ssh);
+            if (WOLFSSH_SESSION_SUBSYSTEM == wolfSSH_GetSessionType(ssh)
+                               && (WMEMCMP(cmd, "sftp", sizeof("sftp")) == 0)) {
```
Comment (ejohnstown) [inline src/ssh.c]: Could this be a `strcpy()` or `strncpy()`?

File: src/ssh.c [hunk 4]
```diff
@@ -608,6 +620,13 @@ int wolfSSH_TriggerKeyExchange(WOLFSSH* ssh)
     return ret;
 }
 
+#if 0
+static void DumpBuf(byte* buf, word32 bufSz) {
```
Comment (ejohnstown) [inline src/ssh.c]: Can this be removed? We also have DumpOctetString available in internal.c.

File: src/ssh.c [hunk 5]
```diff
@@ -839,7 +907,7 @@ int wolfSSH_KDF(byte hashId, byte keyId,
 
 WS_SessionType wolfSSH_GetSessionType(const WOLFSSH* ssh)
 {
-    WLOG(WS_LOG_DEBUG, "Entering wolfSSH_GetSessionCommand()");
+    WLOG(WS_LOG_DEBUG, "Entering wolfSSH_GetSessionType()");
```
Comment (ejohnstown) [inline src/ssh.c]: This is going to conflict with the master. Aaron fixed this string in PR #66.

**Comment by ejohnstown** [comment]
I tried changing directory to one with a space in the name: `cd Virtual\ Machines`. It looks like you don't support backslash escaping. Also, the client got lost, I couldn't change directories anywhere else.

**Comment by JacobBarthelmeh** [comment]
I see, the client side has issues when connected to a server and typing the command "cd Virtual\ Machines". OpenSSH client connected to wolfSSH SFTP server does not. Will look into the client side issue.

**Pull request merged**